code/presentation/mb-client/src/screens/home/namaz-timings.tsx

```diff
@@ -0,0 +1,69 @@
+import { useRoute } from '@react-navigation/native';
+import React, { useState } from 'react';
+
+import { fetchNamazTimings } from '@/core/supabase';
+import type { RouteProp } from '@/navigation/types';
+import { Text, View } from '@/ui';
+
+export const NamazTimingsScreen = () => {
+  const { params } = useRoute<RouteProp<'NamazTimingsScreen'>>();
+  const selectedMasjidId = params.selectedMasjidId;
+
+  const [namazTimings, setNamazTimings] = useState<any[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [fetchExecuted, setFetchExecuted] = useState(false); // Flag to track if fetch operation is executed
+
+  // Call fetchNamazTimings when selectedMasjidId changes and fetchExecuted is false
+  if (selectedMasjidId && !fetchExecuted) {
+    fetchNamazTimings(selectedMasjidId)
+      .then((data) => {
+        setNamazTimings(data);
+      })
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+      .catch((error) => {
+        setError(error);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+    setFetchExecuted(true); // Set flag to true after fetch operation is executed
```

Comment:
we should use, useQuery to fetch and maintain that state, not callbacks

Reply:
Done